app/models/patient.rb

```diff
@@ -698,6 +698,10 @@ def send_assessment(send_now: false)
       end
     end
 
+    # Check last_assessment_reminder_sent before enqueueing to cover potential race condition of multiple reports
+    # being sent out for the same monitoree.
+    return unless last_assessment_reminder_sent <= 12.hours.ago || last_assessment_reminder_sent.nil?
```

Comment:
1. Probably worth moving `last_assessment_reminder_sent_eligible` as a method on the Patient model so you can do the check here as well as in the `PatientMailer` while only needing one method.
2. Also, it would be nice if we could bypass this if `send_now` is true, which is helpful when testing.

Reply:
1. That's a great idea! Made this change.
2. Agreed - I could see that being handy in console sessions. Made this change.